python/tank/pipelineconfig.py

```diff
@@ -83,59 +81,118 @@ def __init__(self, pipeline_configuration_path):
         self._roots = pipelineconfig_utils.get_roots_metadata(self._pc_root)
 
         # get the project tank disk name (Project.tank_name), stored in the PC metadata file.
-        data = pipelineconfig_utils.get_metadata(self._pc_root)
-        if data.get("project_name") is None:
-            raise TankError("Project name not defined in config metadata for config %s! "
-                            "Please contact support." % self._pc_root)
-        self._project_name = data.get("project_name")
+        pipeline_config_metadata = self._get_metadata()
+        self._project_name = pipeline_config_metadata.get("project_name")
+        self._project_id = pipeline_config_metadata.get("project_id")
+        self._pc_id = pipeline_config_metadata.get("pc_id")
+        self._pc_name = pipeline_config_metadata.get("pc_name")
+        self._published_file_entity_type = pipeline_config_metadata.get("published_file_entity_type", "TankPublishedFile")        
+        self._use_shotgun_path_cache = pipeline_config_metadata.get("use_shotgun_path_cache", False)
+
+        # figure out if to use the global bundle cache or the local pc 'install' cache
```

Comment:
minor: For people new to Toolkit, browsing through the codebase, I think it might be better to use a spelled out "pipeline config" rather than the shortened "pc". Very minor, but I think it would help with readability in the long run.